### DISCUSSION TITLE
systemd service does not work with www-data user: onedrive-www-data@.service

### DISCUSSION THREAD

**Discussion by GrzegorzSlowikowski** [abraunegg/onedrive#1473] [Q&A] [answered]
SharePoint Folder Sync works fine. I am using a non-root user and individual configuration.

>user@host:~$ onedrive --confdir="/home/user/.config/user-config" --monitor

I cannot run the program as a service on a dedicated user.  
I have got such a answer.

> user@host:~$ ExecStart=/usr/local/bin/onedrive --monitor --confdir="/home/user/.config/user"
> -bash: --monitor: command not  found

The path to config is valid and service is  already set as   onedrive-user@.service   
What is the problem

**Comment by abraunegg** [comment]
@GrzegorzSlowikowski 
Have you 'manually' modified the systemd files for some reason?

**Comment by GrzegorzSlowikowski** [comment]
Not at all.  But may be a problem is that my <user> has a specific name. 
**user = www-data**
Therefore, the website has a specific name.   **onedrive-www-data@.service**

Could this be the cause?

**Comment by abraunegg** [comment, marked as answer]
This to me sounds more like an Operating System issue / systemd service issue then. The service is unable to parse properly the user portion of the tokens as parsed. This is what is failing you, not the 'onedrive' client. This is not something I can fix.

Suggest you open a trouble ticket with:

* Your Linux Operating System support folk
* The systemd development team

Additional references:
* https://unix.stackexchange.com/questions/423632/systemctl-user-not-available-for-www-data-user
* https://serverfault.com/questions/241330/starting-daemons-as-www-data

**Comment by GrzegorzSlowikowski** [comment]
Now I'm sure it's not an operating system problem. I installed a completely new system for testing. I used a standard user. 
When starting the service manually, it works. Both --synchronize and --monitor
```
_gslowikowski@debian-cli:~$ onedrive --confdir="/home/gslowikowski/.config/gsconfig" --synchronize
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Uploading differences of /home/gslowikowski/pliki
Uploading new items of /home/gslowikowski/pliki__
```

```
_gslowikowski@debian-cli:~$ onedrive --confdir="/home/gslowikowski/.config/gsconfig" --monitor
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 300__
```
```
Onedrive@user.service attempt to start service does not recognize switch --monitor 

_gslowikowski@debian-cli:~$ ExecStart=onedrive --monitor --confdir="/home/gslowikowski/.config/gsconfig"
-bash: --monitor: command not found_
```

**Comment by abraunegg** [comment]
@GrzegorzSlowikowski 
This is *not* a problem with this client. Please stop thinking this is the case.

Systemd is what is failing you here ... not this client.

This issue is this:
* systemd not parsing the required user id correctly
* your OS implementing systemd the way it has

It is systemd that is complaining that it cannot execute the command, not this client.

Raise a bug / issue with your OS support and systemd
